crates/adapters/src/server/mod.rs

```diff
@@ -97,17 +121,27 @@ fn http_resp_from_error(error: &AnyError) -> HttpResponse {
                 config_error: ConfigError::UnknownOutputStream { .. },
             } => HttpResponse::NotFound(),
             ControllerError::Config { .. } => HttpResponse::BadRequest(),
+            ControllerError::ParseError { .. } => HttpResponse::UnprocessableEntity(),
```

Comment:
I'm not sure exactly what throws `ParseError`, but `HttpResponse::UnprocessableEntity` is for when the request is well-formed, which I suspect is not the case here. From the sound of it, you probably want `HttpResponse::BadRequest`?

Reply:
I think you're right. This error gets generated when parsing the streaming payload, e.g., `csv` file data.